charms/worker/k8s/src/config/extra_args.py

```diff
@@ -58,37 +70,35 @@ def craft(
             The configuration object to be updated with extra arguments.
         cluster_name (str): the name of the cluster to override in the extra arguments.
         node_ips (list[str]): the IP address of the node to override in the extra arguments.
+        datastore(Optional[str]): the name of the datastore.
     """
+
+    def update_args(key: str, extra_args: dict):
+        """Parse and update arguments."""
+        cmd = _parse(src.get(key, ""))
+        cmd.update(extra_args)
+        return cmd
+
     if isinstance(dest, (BootstrapConfig, ControlPlaneNodeJoinConfig)):
-        cmd = _parse(src["kube-apiserver-extra-args"])
-        dest.extra_node_kube_apiserver_args = cmd
+        dest.extra_node_kube_apiserver_args = _parse(src["kube-apiserver-extra-args"])
 
-        cmd = _parse(src["kube-controller-manager-extra-args"])
-        if cluster_name:
-            cmd.update(**{"--cluster-name": cluster_name})
-        else:
-            cmd.pop("--cluster-name", None)
-        dest.extra_node_kube_controller_manager_args = cmd
-
-        cmd = _parse(src["kube-scheduler-extra-args"])
-        dest.extra_node_kube_scheduler_args = cmd
-
-        cmd = _parse(src["datastore-extra-args"])
-        match src["bootstrap-datastore"]:
-            case literals.DATASTORE_TYPE_K8S_DQLITE:
-                dest.extra_node_k8s_dqlite_args = cmd
-            case literals.DATASTORE_TYPE_ETCD:
-                dest.extra_node_etcd_args = cmd
-
-    cmd = _parse(src["kube-proxy-extra-args"])
-    dest.extra_node_kube_proxy_args = cmd
-
-    cmd = _parse(src["kubelet-extra-args"])
-    if node_ips:
-        cmd.update(**{"--node-ip": ",".join(node_ips)})
-    else:
-        cmd.pop("--node-ip", None)
-    dest.extra_node_kubelet_args = cmd
+        controller_args = {"--cluster-name": cluster_name} if cluster_name else {}
+        dest.extra_node_kube_controller_manager_args = update_args(
+            "kube-controller-manager-extra-args", controller_args
+        )
```

Comment:
i think there's a regression here.  We don't want the users to set the `--cluster-name` field.  The charm sets it -- or the charm prevents it